### PR TITLE
Use tls by default for application ingress

### LINKED ISSUE #46: Apps not available via HTTPS
## Summary
HTTPS traffic does not reach apps running on cf-for-k8s.

## Repro Steps
1. Deploy cf-for-k8s and log in via the cf CLI
1. Enable feature flag 'diego_docker'
1. Target a space and push a docker app
1. Attempt to curl the route provided in the push output and observe it is only available over unencrypted http.
```
> cf push myapp -o cloudfoundry/diego-docker-app

> curl -i http://myapp.<app-domain>
HTTP/1.1 200 OK
date: Wed, 26 Feb 2020 00:21:36 GMT
content-length: 1
content-type: text/plain; charset=utf-8
x-envoy-upstream-service-time: 3
server: istio-envoy

0

> curl -i https://myapp.<app-domain> -k
HTTP/2 404
date: Wed, 26 Feb 2020 00:21:18 GMT
server: istio-envoy
```

## Context
Currently, there are two istio ingress gateways: one for the cf-system components and one for cf-workloads. The ingress gateway for cf-workloads namespace only configures a rule for unencrypted http traffic to port 80. Meanwhile the system component ingress gateway only listens for encrypted https traffic on port 443. So any encrypted https traffic to port 443 only matches the routing rule on the cf-system gateway and receives an istio 404 response. We expect to work with the cf-for-k8s networking team to resolve this issue.

### PR DESCRIPTION
**Pull request by XanderStrike**
- This enables tls by default for the workload domain
- It updates generate values, but does not enforce different apps and system domains
  - When app and system domain are the same, system domain is used (this is what generate values does)
  - When app and system domain are different, separate certs are used

**TL;DR: This implementation no longer requires app and system domains to be different.**

fixes #46

**Acceptance Steps**

```
GIVEN a kubernetes cluster
WHEN I deploy cf-for-k8s
AND push an application
THEN I can reach that application via https
```

@jrussett @jenspinney @ndhanushkodi @tcdowney @kauana

### CODE REVIEW

File: config/_ytt_lib/github.com/cloudfoundry/cf-k8s-networking/config/cfroutesync/istio-gateway.yaml [hunk 1]
```diff
@@ -1,17 +0,0 @@
-#@ load("@ytt:data", "data")
----
```
Comment (tcdowney) [inline config/_ytt_lib/github.com/cloudfoundry/cf-k8s-networking/config/cfroutesync/istio-gateway.yaml:2]: Why are we switching the apps over to using the `cf-system` Gateway?

I think we'll also need to update these docs: https://github.com/cloudfoundry/cf-k8s-networking/blob/master/doc/managing-certificates-for-application-ingress.md

Comment (XanderStrike) [reply config/_ytt_lib/github.com/cloudfoundry/cf-k8s-networking/config/cfroutesync/istio-gateway.yaml:2]: Having multiple gateways doesn't buy us anything. If two gateways have conflicting configuration Istio will silently accept them and then generate invalid configuration to ship to the ingress gateways. Better in my opinion to have a single gateway as the source of truth where we can see if we've accidentally tried to make two `servers` with the same wildcard host.

Also, app domains aren't necessarily tied to a given space or org, so in the ideal future where spaces are mapped to namespaces it would leave us with a complicated tangle of gateways.

File: config/external-routing.yml [hunk 2]
```diff
@@ -24,32 +36,8 @@ metadata:
     kapp.k14s.io/change-group: istio.io/ingressgateway
 
 #! this gateway is used to expose CF services in the cluster, externally
```
Comment (jamespollard8) [inline config/external-routing.yml:38]: might be nice to include this in-line comment at the top of the gateway.lib.yml file too

File: config/gateway.lib.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+kind: Gateway
+metadata:
+  name: istio-ingressgateway
+  namespace: #@ system_namespace
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - hosts:
+    #! TODO: restrict this
```
Comment (jamespollard8) [inline config/gateway.lib.yml:12]: Are you planning to address this TODO as part of this PR? If not, do you have a story to track this?

Comment (jrussett) [reply config/gateway.lib.yml:12]: @jamespollard8, @XanderStrike and I wrote [this story](https://www.pivotaltracker.com/story/show/172662780) in our tracker to figure out how we wanted to deal with this `TODO`.  We did not plan on addressing the comment in this PR.

Comment (jamespollard8) [reply config/gateway.lib.yml:12]: Ah, on first pass I didn't catch that this TODO was moved, not created

Comment (jenspinney) [reply config/gateway.lib.yml:12]: Also, just in case it wasn't clear, the TODO wasn't added in this PR. This was in the old external-routing.yml that this was extracted from.

(I believe I added it back in the early days of cf-for-k8s when we were just racing to get anything working at all).
